tool/src/main/java/org/wildfly/security/tool/CredentialStoreCommand.java

```diff
@@ -343,9 +506,64 @@ private Credential createCredential(final String secret, String entryType) {
         }
     }
 
+    private KeyPairCredential createKeyPairCredential(String algorithm, int size) throws NoSuchAlgorithmException, NoSuchProviderException {
+        KeyPairGenerator keyPairGenerator;
+
+        switch (algorithm) {
+            case Pem.RSA_ALGORITHM: {
+                size = (size < 512 | size > 2048) ? 2048 : size;
```

Comment:
Would be good to add a link to where these bounds are from and also update the proposal to match the numbers here.

Reply:
Fixed, thanks!